### DISCUSSION TITLE
medical-docs: Broken package and package vetting and versioning

### DISCUSSION THREAD

**Discussion by reagle** [espanso/espanso#1592] [Q&A] [answered]
I'm learning espanso and medical-docs looked like a good exemplar of espanso's functionality but it doesn't work. When I invoke [`:chestpain`](https://github.com/wressl/medical-docs/blob/master/medical-docs/0.1.3/cardio.yml) I get the following error. This also made me wonder if this package is up to date? Are packages versioned and/or vetted?

```
...
17:23:19 [daemon(11992)] [INFO] system info: Darwin v13.3.1 - kernel: 22.4.0
17:23:19 [daemon(11992)] [INFO] watching for changes in path: "/Users/reagle/Library/Application Support/espanso"
17:23:19 [daemon(11992)] [INFO] espanso version: 2.1.8
...
17:24:22 [worker(12123)] [INFO] using CocoaSource
17:24:22 [worker(12123)] [INFO] monitoring the status of secure input
17:24:22 [worker(12123)] [INFO] monitoring the status of the daemon process
17:24:22 [worker(12123)] [INFO] using MacInjector
17:24:22 [worker(12123)] [INFO] using CocoaClipboard
17:24:35 [worker(12123)] [ERROR] could not resolve variable "ap"
17:24:35 [worker(12123)] [ERROR] error during rendering: rendering error

Caused by:
    missing variable: `ap`
```

**Comment by smeech** [comment, marked as answer]
Sorry this has hung here for so long. I'm trying to mop up some of the Issues & Discussions.

Did you follow the instructions at https://github.com/wressl/medical-docs/blob/master/README.md and make the additions to your `default.yml`? That's where the global variables for this package appear to be defined. If you have, may I suggest you check the alignment of the code in case that's preventing the variable being picked up.

If you're still having problems, it may be quicker for you to check the fork at the above address and enquire of the author.

**Comment by reagle** [comment]
@smeech thanks for writing; I honestly don't remember -- I've been happily using Espanso in the meantime -- but I suspect you are correct. If so, I think it'd make sense for the package not to be reliant on changes to the users default.yml .

**Reply by smeech** [reply]
Yes, I'm not sure why the author didn't put the `global_vars:` into the top of `package.yml`. It's a package I might have used in my professional career had I not retired before its incipience!

Thank you for taking the trouble to reply after so long. If it's OK with you, I'll close the discussion.
